src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -274,13 +275,36 @@ CHIP_ERROR DiscoveryImplPlatform::Advertise(const OperationalAdvertisingParamete
 
     ReturnErrorOnFailure(MakeInstanceName(service.mName, sizeof(service.mName), params.GetFabricId(), params.GetNodeId()));
     strncpy(service.mType, "_chip", sizeof(service.mType));
-    service.mProtocol      = MdnsServiceProtocol::kMdnsProtocolTcp;
-    service.mPort          = CHIP_PORT;
-    service.mTextEntries   = nullptr;
-    service.mTextEntrySize = 0;
-    service.mInterface     = INET_NULL_INTERFACEID;
-    service.mAddressType   = Inet::kIPAddressType_Any;
-    error                  = ChipMdnsPublishService(&service);
+    service.mProtocol = MdnsServiceProtocol::kMdnsProtocolTcp;
+    service.mPort     = CHIP_PORT;
+    TextEntry crmpRetryIntervalEntries[2];
+    uint32_t crmpRetryIntervalIdle   = CHIP_DEVICE_CONFIG_CRMP_RETRY_INTERVAL_IDLE;
+    uint32_t crmpRetryIntervalActive = CHIP_DEVICE_CONFIG_CRMP_RETRY_INTERVAL_ACTIVE;
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+    if (chip::DeviceLayer::ConnectivityMgr().GetThreadDeviceType() ==
+        chip::DeviceLayer::ConnectivityManager::kThreadDeviceType_SleepyEndDevice)
+    {
+        uint32_t sedPollPeriod;
+        ReturnErrorOnFailure(chip::DeviceLayer::ThreadStackMgr().GetPollPeriod(sedPollPeriod));
```

Comment:
There are many things that bother me:

1. The Poll Period is configurable during the lifetime of the device. I believe we should update CRMP parameters when we update the poll period - though it may be done quite frequently. Currently, in OpenThread AFAIK there is no way to get asynchronous notification about the poll period being changed.
2. Device may change its role dynamically from SED to MED, and in another way around. Similarly, we don't have asynchronous information about that from OT.
3. I would consult the actual values for CRMP with Google (which might be done during review). The current proposal looks fine, another approach could be to use the MLE Timeout instead - though I prefer your way.

Saying all that, i think it's a good start, but we might consider updating the CRMP values when someone tries to change poll interval or change the role using CHIP's Thread API - or make necessary changes in OpenThread.

Reply:
As discussed offline, I created issue describing problem and TODO comment to implement mentioned functionalities in the next iterations.